### DISCUSSION TITLE
Issues with mocking "next/router" after upgrading to Next.js 12.2

### DISCUSSION THREAD

**Discussion by nitintrainline** [vercel/next.js#38335] [Help]
## Problem

Mocking `next/router` in the component's test file doesn't seems to be working after upgrading to **Next.js 12.2**. The tests are failing with the following error message -

<img width="779" alt="Screenshot 2022-07-05 at 15 24 22" src="https://user-images.githubusercontent.com/102592759/177351029-cb8d231e-5e70-45c0-9f0e-afb13be85617.png">

Before upgrading to **Next.js 12.2**, there were no issues with the tests i.e mocking `useRouter` from the dependency `next/router` was working fine.

## How am I mocking the `next/router`?

```ts
jest.mock('next/router');

import Router from 'next/router';
```

```ts
    expect(Router.push).toHaveBeenCalledTimes(1);
    expect(Router.push).toHaveBeenCalledWith({
      pathname: '..',
      query: '...',
    });
```

[Here is an another example](https://stackoverflow.com/questions/72847423/jestnextjs-you-should-only-use-next-router-on-the-client-side-of-your-app) of this issue where the author is describing the same problem after upgrading to **12.2**.

**Comment by GabenGar** [comment]
Does adding [jsdom declaration](https://jestjs.io/docs/configuration#testenvironment-string) at the top of the file help?

**Reply by nitintrainline** [reply]
Thanks @GabenGar.

So I added the following docblock at the top of my test file, and I am still getting the same error 😕 

```ts

/**
 * @jest-environment jsdom
 */

jest.mock('next/router');

import Router from 'next/router';
```
